### PR TITLE
made data pattern consistent across all the page components wrt useFetch

### LINKED ISSUE #37: consistent data pattern across all the page components
Based on our useFetch service, the data pattern needs to be consistent across all page components (this is definitely what we need for a more readable and efficient design), for example:
```
// top level
const {error, loading, data} = useFetch(url)

if (error) return <div className="errorStyles">{error.message}</div>

if (loading && !data) return <div className="loading"><Spinner /></div>

if (!data) return <div>No data for the request</div>

// finally, shows data if everything's alright
return <div>{data.data}</div>
```

### PR DESCRIPTION
**Pull request by akshatgarg12**
fixes #37

### CODE REVIEW

**Comment by aquibbaig** [comment]
Let me know when this is ready for a review

**Comment by akshatgarg12** [comment]
> Let me know when this is ready for a review

Ready from my side

**Review by aquibbaig — COMMENTED** [review]
You are not adding any dependency, so do not make changes to package.json

File: src/components/GraphWrapper/graphWrapper.test.tsx [hunk 1]
```diff
@@ -12,11 +12,8 @@ import { queryEntities } from "../../services/queryEntity";
 import constants from "../../utils/constants";
 import { useGlobalStore } from "../../store/global";
 
-const {
-  defaultStartTimestamp,
-  defaultStepValue,
-  defaultEndTimestamp,
-} = constants;
+const { defaultStartTimestamp, defaultStepValue, defaultEndTimestamp } =
```
Comment (aquibbaig) [inline src/components/GraphWrapper/graphWrapper.test.tsx:15]: shouldn't be a part of the changes, please enable prettier on your system to auto save formattings

File: src/components/GraphWrapper/index.tsx [hunk 2]
```diff
@@ -20,11 +20,8 @@ interface PageProps {
 
 const GraphWrapper: React.FC<PageProps> = (props: PageProps) => {
   const { selectedRoutePath } = props;
-  const {
-    selectedStartTimestamp,
-    selectedEndTimestamp,
-    selectedStepValue,
-  } = useTimeQuerierStore();
+  const { selectedStartTimestamp, selectedEndTimestamp, selectedStepValue } =
```
Comment (aquibbaig) [inline src/components/GraphWrapper/index.tsx:23]: shouldn't be a part of the changes

File: src/components/MachineSelector/index.tsx [hunk 3]
```diff
@@ -1,5 +1,11 @@
 import React, { useEffect } from "react";
-import { Select, Alert, AlertIcon, VStack } from "@chakra-ui/react";
+import {
```
Comment (aquibbaig) [inline src/components/MachineSelector/index.tsx:2]: these formatting changes look good 👍

File: src/components/TimeQuerier/index.tsx [hunk 4]
```diff
@@ -20,11 +20,8 @@ import { useTimeQuerierStore } from "../../store/timeQuerier";
 
 const TimeQuerier: React.FC = () => {
   const styles = useStyleConfig("DateTime", {});
-  const {
-    selectedStartTimestamp,
-    selectedEndTimestamp,
-    changeTimeQuerier,
-  } = useTimeQuerierStore();
+  const { selectedStartTimestamp, selectedEndTimestamp, changeTimeQuerier } =
```
Comment (aquibbaig) [inline src/components/TimeQuerier/index.tsx:23]: the older changes were fine

File: src/theme/components/DateTime.ts [hunk 5]
```diff
@@ -26,9 +26,10 @@ export const DateTime = {
     },
     ".rdtPicker td.rdtDay:hover,.rdtPicker thead tr:first-of-type th:hover\
     ,.rdtPicker .rdtTimeToggle:hover,.rdtCounter .rdtBtn:hover\
-    ,td.rdtMonth:hover, td.rdtYear:hover,.rdtSwitch:hover": {
-      bg: mode("#eee", "darkPrimary")(props),
-    },
+    ,td.rdtMonth:hover, td.rdtYear:hover,.rdtSwitch:hover":
```
Comment (aquibbaig) [inline src/theme/components/DateTime.ts:29]: revert

File: src/utils/useFetch.ts [hunk 6]
```diff
@@ -74,6 +74,7 @@ const useFetch = <T = unknown>(
           if (cancelRequest.current) return;
 
           dispatch({ type: "success", payload: response.data });
+          // @ts-ignore
```
Comment (aquibbaig) [inline src/utils/useFetch.ts:77]: why was this change introduced here?

**Review by aquibbaig — CHANGES_REQUESTED** [review]
Thanks for the PR and sorry for the late review, I've added some comments. These changes more or less look great, some small changes and we're good to go

File: src/utils/useFetch.ts (hunk 6 above)
Comment (akshatgarg12) [reply src/utils/useFetch.ts:77]: this is not required , I'll remove this

**Comment by akshatgarg12** [comment]
@aquibbaig all of the formatting changes are introduced while commiting the code, as it runs yarn prettier -fix, how should i revert these changes?

File: src/utils/useFetch.ts (hunk 6 above)
Comment (aquibbaig) [reply src/utils/useFetch.ts:77]: Please do

**Comment by aquibbaig** [comment]
> @aquibbaig all of the formatting changes are introduced while commiting the code, as it runs yarn prettier -fix, how should i revert these changes?

If it is a part of basic `git commit`  workflow, then lets not change that, we will redefine these rules as a part of a new prettier configuration

File: src/utils/useFetch.ts (hunk 6 above)
Comment (akshatgarg12) [reply src/utils/useFetch.ts]: done

**Review by aquibbaig — APPROVED** [review]

**Pull request merged**